lock_api/src/mutex.rs

```diff
@@ -681,7 +681,18 @@ unsafe impl<'a, R: RawMutex + 'a, T: ?Sized + 'a> StableAddress for MutexGuard<'
 #[must_use = "if unused the Mutex will immediately unlock"]
 pub struct ArcMutexGuard<R: RawMutex, T: ?Sized> {
     mutex: Arc<Mutex<R, T>>,
-    marker: PhantomData<R::GuardMarker>,
+    // n.b. notgull: add a ptr here to prevent the MutexGuard
+    // from being Sync unless it is Sync itself.
+    marker: PhantomData<(R::GuardMarker, *const ())>,
+}
+
+unsafe impl<R: RawMutex, T: Send + Sync + ?Sized> Send for ArcMutexGuard<R, T> where
+    R::GuardMarker: Send
+{
+}
+unsafe impl<R: RawMutex, T: Send + Sync + ?Sized> Sync for ArcMutexGuard<R, T> where
```

Comment:
```suggestion
unsafe impl<R: RawMutex, T: Sync + ?Sized> Sync for ArcMutexGuard<R, T> where
```
Same here I think we only require `T: Sync` here.